src/Agent.Plugins/RepositoryPlugin.cs

```diff
@@ -133,13 +137,22 @@ public override async Task RunAsync(AgentTaskPluginExecutionContext executionCon
                     throw new ArgumentException($"Input path '{path}' should resolve to a directory under '{buildDirectory}', current resolved path '{expectRepoPath}'.");
                 }
             }
+            else if (HasMultipleCheckouts(executionContext))
```

Comment:
Will it always be the case that path is null when HasMultipleCheckouts is true?

Reply:
no. If the path is set on the checkout task, we use that no matter what. Here we are differentiating between how we want to default the path if it is not set. Multi-repo changes the default.

Reply:
Nit: Can we add this as a comment in code to make it clear?